lib/elastomer/client/scroller.rb

```diff
@@ -45,7 +45,13 @@ def scroll( query, opts = {} )
     #
     # Returns a new Scroller instance
     def scan( query, opts = {} )
-      opts = opts.merge(:search_type => "scan")
+      if query.nil?
+        query = {}
+      elsif query.is_a? String
+        query = MultiJson.load(query)
+      end
+      query = query.merge(:sort => [:_doc])
```

Comment:
One thing I was thinking about with this change was to fail fast if the passed in query had a sort. What do you think? 

The nice thing about that is if you were expecting sorting, it will complain loudly.

Reply:
That makes a lot of sense. I understand you correctly you are proposing to raise an exception right away if the query contains a `:sort` already.